src/main/kotlin/com/cosmotech/api/utils/ResourceScanner.kt

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+package com.cosmotech.api.utils
+
+import com.cosmotech.api.exceptions.CsmAccessForbiddenException
+import java.io.BufferedInputStream
+import java.util.zip.ZipEntry
+import java.util.zip.ZipInputStream
+import org.apache.tika.config.TikaConfig
+import org.apache.tika.metadata.Metadata
+import org.apache.tika.metadata.TikaCoreProperties
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.core.io.Resource
+import org.springframework.stereotype.Component
+
+private const val ZIP_MIME_TYPE = "zip"
+private const val ENTRY_NAME_UNKNOWN = "Unknown"
+
+@Component
+class ResourceScanner {
+  private val logger: Logger = LoggerFactory.getLogger(this::class.java)
+
+  fun scanMimeTypes(file: Resource, authorizedMimeTypes: List<String>) {
+    val tika = TikaConfig()
+    val metadata = Metadata()
```

Comment:
It's better to extract this part of code to something like this:
```
fun getMimeType(fileName: String, inputStream: InputStream): MimeType {
   // Maybe check if InputStream is BufferedInputStream (in case of file.inputStream)
    val metadata = Metadata()
    metadata.set(TikaCoreProperties.RESOURCE_NAME_KEY, fileName)
    var mimetype = tika.detector.detect(inputStream, metadata)
    return tika.detector.detect(bufferedStream, metadata)
}
```